apps/omg_watcher/lib/omg_watcher/api/status_cache/storage.ex

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.API.StatusCache.Storage do
+  @moduledoc """
+  Watcher status API storage
+  """
+
+  alias OMG.Eth
+  alias OMG.Eth.Client
+  alias OMG.Eth.Configuration
+  alias OMG.Eth.RootChain
+  alias OMG.RootChainCoordinator
+  alias OMG.State
+  alias OMG.Utils.HttpRPC.Encoding
+  alias OMG.Watcher.BlockGetter
+  alias OMG.Watcher.Event
+  alias OMG.Watcher.ExitProcessor
+
+  @type t() :: %{
+          last_validated_child_block_number: non_neg_integer(),
+          last_validated_child_block_timestamp: non_neg_integer(),
+          last_mined_child_block_number: non_neg_integer(),
+          last_mined_child_block_timestamp: non_neg_integer(),
+          last_seen_eth_block_number: non_neg_integer(),
+          last_seen_eth_block_timestamp: non_neg_integer(),
+          eth_syncing: boolean(),
+          byzantine_events: list(Event.t()),
+          in_flight_exits: ExitProcessor.Core.in_flight_exits_response_t(),
+          contract_addr: binary,
+          services_synced_heights: RootChainCoordinator.Core.ethereum_heights_result_t()
+        }
+  @doc """
+  This gets periodically called (defined by Ethereum height change).
+  """
+  def update_status(state, key, eth_block_number) do
+    {:ok, status} = get_status(eth_block_number)
+    :ets.insert(state, {key, status})
+  end
+
+  # Returns status of the watcher. Status consists of last validated child block number,
+  # last mined child block number and it's timestamp, and a flag indicating if watcher is syncing with Ethereum.
+
+  # This function calls into a number of services (internal and external), collects the results.
+  # If any of the underlying services are unavailable, it will crash
+  @spec get_status(non_neg_integer()) :: {:ok, t()}
+  defp get_status(eth_block_number) do
+    {:ok, eth_block_timestamp} = Eth.get_block_timestamp_by_number(eth_block_number)
+    eth_syncing = syncing?()
+
+    validated_child_block_number = get_validated_child_block_number()
+    contracts = Configuration.contracts()
+    contract_addr = contract_map_from_hex(contracts)
+
+    mined_child_block_number = RootChain.get_mined_child_block()
+
```

Comment:
why the newlines? Is this the formatter?

Reply:
formatter is ass